### DISCUSSION TITLE
How to build and load the firmware

### DISCUSSION THREAD

**Discussion by boonsp** [jgressmann/supercan#14] [Q&A]
I was looking for a way to analyse a can bus of a Bosch base bike and came across this project. So I a bought a [Adafruit Feather M4 CAN Express] but I am failing the instructions how to build the firmware (which IDE to use) and how to flash it into the board?
It seems that this project supposes that all this knowledge is already available, but in my case it is not, an I guess a lot of people are somewhat in the same situation. So how cab I get out of this deadlock situation? A step by step guideline would be ideal, but I guess that I am expecting too much here.

**Comment by jgressmann** [comment]
There used to be build instructions, I guess this is on me. 
You can take the pre-built firmware file from the [releases page of this project](https://github.com/jgressmann/supercan/releases/). Unpack the firmware archive, find the file for your board and follow the instructions provided in the readme.

**Comment by boonsp** [comment]
Thanks for the swift answer. 
I wil try the prebuilt FW.

In addition herewith my best wishes. I wish you a splendid 2024.

**Comment by zerotoney** [comment]
sorry, can I have the instruction and source code for the firmware? kind of lost on that. Thank you.

**Reply by jgressmann** [reply]
The sources for the firmware live [here](https://github.com/jgressmann/tinyusb/tree/master/examples/device/supercan).
